Core/include/Acts/Detector/detail/BlueprintDrawer.hpp

```diff
@@ -0,0 +1,163 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/Blueprint.hpp"
+#include "Acts/Utilities/BinningData.hpp"
+
+namespace Acts {
+
+namespace Experimental {
+
+namespace detail {
+namespace BlueprintDrawer {
+
+/// @brief Nested options struct for the drawer
+struct Options {
+  struct Node {
+    /// ROOT node definitions
+    std::string shape = "circle";
+    std::string color = "darkorange";
+
+    /// Font properties
+    std::string face = "sans-serif";
+    int labelText = 12;
+    int infoText = 10;
+
+    /// Info properties
+    int precision = 1;
+  };
+
+  /// @brief The name of the graph
+  std::string graphName = "blueprint";
+
+  // Main node types to come
+  Node root = Node{};
+  Node branch = Node{"diamond", "white"};
+  Node leaf = Node{"box", "darkolivegreen1"};
+  Node gap = Node{"box", "darkolivegreen3"};
+
+  // Sub node types to come
+  Node shape = Node{"cylinder", "lightgrey"};
+  Node virtualShape = Node{"cylinder", "white"};
+  Node internals = Node{"doubleoctagon", "cadetblue1"};
+  Node geoID = Node{"box", "azure"};
+  Node roots = Node{"box", "darkkhaki"};
+};
+
+/// @brief Geneate the shape string
+/// @param s the shape of the object
+/// @param c the color of the object
+/// @return a string with the shape and color
+std::string shapeStr(const Options::Node& node) {
```

Comment:
Not super critical, but these could go into a `.cpp` file.

Reply:
done.